### DISCUSSION TITLE
Link in CommandItem

### DISCUSSION THREAD

**Discussion by NixBiks** [shadcn-ui/ui#1735] [Q&A]
I just checked the source code for the documentation page, i.e. for the [command-menu](https://github.com/shadcn-ui/ui/blob/43c4023ed8f8c150cfbd01f5d7fe2773938b25b3/apps/www/components/command-menu.tsx#L78). It looks like this

```ts
<CommandItem
    key={navItem.href}
    value={navItem.title}
    onSelect={() => {
    runCommand(() => router.push(navItem.href as string))
    }}
>
    <FileIcon className="mr-2 h-4 w-4" />
    {navItem.title}
</CommandItem>
```

So for navigating we are using `router.push` instead of using `Link` from `next/link`. Isn't that suboptimal? I assume you'd prefer to have prefetching which comes for free with `Link` but is that even possible with `CommandItem`? Alternatively you can do `router.prefetch` I guess. See here for [more details](https://nextjs.org/docs/app/building-your-application/routing/linking-and-navigating#how-routing-and-navigation-works). I realise it isn't a big deal for this site (probably) but in other cases it might be.

**Comment by Yhozen** [comment]
👀

**Comment by Yhozen** [comment]
```tsx
import Link from 'next/link';
import { CommandItem } from '@/components/ui/command';

export function CommandMenu() {
  return (
    <CommandItem asChild>
      <Link href="/dashboard">Dashboard</Link>
    </CommandItem>
  );
}
```

**Reply by fmarkulin** [reply]
This doesn't trigger the routing via the Enter key, which is the expected interaction with Command Dialog

**Reply by DanielRamosAcosta** [reply]
@fmarkulin did you solved this problem?

**Comment by zramdeen** [comment]
Not sure if this is efficient but you could just create the link and press it with the `onSelect` manually.

```typescript
<CommandItem
  key={item.id}
  onSelect={() => {
    const link = document.getElementById(
      `page-link-${item.title}`
    )
    link?.click()
  }}
  asChild
>
  <Link id={`page-link-${item.title}`} href={item.url}>
    {item.title}
  </Link>
</CommandItem>
```
